src/main/java/net/openhft/chronicle/queue/impl/table/SingleTableStore.java

```diff
@@ -50,6 +50,13 @@
 
 public class SingleTableStore<T extends Metadata> extends AbstractCloseable implements TableStore<T> {
     public static final String SUFFIX = ".cq4t";
+    private static final int EXCLUSIVE_LOCK_SIZE = 1;
+    /**
+     * We need to be able to acquire an "exclusive" lock while fine-grained long-running locks are being held.
+     * For this reason the "exclusive" lock doesn't lock the whole file, but as long as everyone agrees
+     * on what section constitutes an "exclusive" lock this should be fine.
+     */
+    private static final long EXCLUSIVE_LOCK_START = Long.MAX_VALUE - EXCLUSIVE_LOCK_SIZE;
```

Comment:
I like this - all the other locks around long values will all have low offsets, and it is clear from looking at the offsets that the "exclusive" lock is a different beast. Nice intentional code :)